pkg/processor/metricfrequencyprocessor/metriccache.go

```diff
@@ -0,0 +1,55 @@
+package metricfrequencyprocessor
+
+import (
+	"github.com/patrickmn/go-cache"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"time"
+)
+
+type DataPoint struct {
+	Timestamp pdata.Timestamp
+	Value     float64
+}
+
+type MetricCache struct {
+	internalCaches map[string]*cache.Cache
```

Comment:
Why not use e.g. `in64` as key type? Can't find a proper blog post/article about it now but it will be more efficient. Then you could use `.AsTime().UnixNano()` instead of `.String()` on the timestamp

Reply:
I keep a cache for each metric. This `string` refers to metric name, not timestamps.

Reply:
Ok, I got mixed up with reading about timestamps elsewhere 🤦